packages/chat_repository_interface/lib/src/models/message_model.dart

```diff
@@ -27,6 +49,7 @@ class MessageModel {
         imageUrl: map["imageUrl"],
         timestamp: DateTime.fromMillisecondsSinceEpoch(map["timestamp"]),
         senderId: map["senderId"],
+        status: MessageStatus.tryParse(map["status"]) ?? MessageStatus.sent,
```

Comment:
Couldn't this make for some really wrong information. In my opinion it's weird to say that when this value is empty or wrong to just set it to the positive and expected value.

Shouldn't a unknown enum value be right here?

Just asking not demanding this or anything.

Reply:
Ideally, yes. But we want to be backwards compatible so that any previous messages would automatically be interpreted as sent, as that is what they always are. 

In the end, the fromMap is really a helper for custom implementation, and ideally should be replaced in whichever domain is implementing the chat.